Respect user-supplied CFLAGS and CXXFLAGS, and do not try to use -O9
which clang does not like.
--- configure.ac.orig	2002-08-20 00:58:13.000000000 -0500
+++ configure.ac	2016-10-24 09:19:02.000000000 -0500
@@ -82,8 +82,8 @@
 dnl check gcc
 dnl
 if test -n "$GCC"; then
-   CFLAGS="-Wall -O9 -funroll-all-loops -finline -ffast-math"
-   CXXFLAGS="-Wall -O9 -funroll-all-loops -finline -ffast-math"
+   CFLAGS="$CFLAGS -Wall -funroll-all-loops -finline -ffast-math"
+   CXXFLAGS="$CXXFLAGS -Wall -funroll-all-loops -finline -ffast-math"
 fi
 
 # On Intel systems with gcc, we may need to compile with -mieee-fp to
